### PR TITLE
Replace custom useQuery util with TanStack useQuery

### PR DESCRIPTION
**Pull request by amjithtitus09**
## Proposed Changes

- Fixes #issue_number
- Replaced custom useQuery util with tanStack useQuery, along with a wrapper for backward compatibility
- Added documentation for tanStack usage as well as wrapper usage

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — APPROVED** [review]
This approach LGTM. 

Just something to note, our `request` function does perform query retries by itself. And in this implementation Tanstack's Query Client has retry configured to 1. 

We could modify our `request` to not perform retries.

https://github.com/ohcnetwork/care_fe/blob/b25ecd001b117624cb4b279f7078232ea2f4fdb4/src/Utils/request/request.ts#L53

File: src/Utils/request/useTanStackQueryInstead.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  const refetch = async (
+    overrides?: QueryOptions<TData>,
+  ): Promise<RequestResult<TData>> => {
+    const resolvedOptions = overrides
+      ? mergeRequestOptions(options || {}, overrides)
+      : options;
+
+    // Directly fetch data with resolved options
+    return await request(route, resolvedOptions);
+  };
```
Comment (rithviknishad) [inline src/Utils/request/useTanStackQueryInstead.ts]: When calling `refetch` in this implementation, we are bypassing tanstack's useQuery invocation made above altogether. This means the returned `data`, `res`, etc. by `useTanStackQueryInstead` is only of the first request and not of subsequent requests.

Hence the `refetch` called for the current auth user when signIn is invoked, the original states aren't updated. Same for signOut and other similar usages in CARE.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Utils/request/useTanStackQueryInstead.ts (hunk 1 above)
Comment (rithviknishad) [reply src/Utils/request/useTanStackQueryInstead.ts]: Let's instead use tanstack's useQuery's `refetch` itself instead?

File: src/Utils/request/useTanStackQueryInstead.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  options?: QueryOptions<TData>,
+) {
+  const { data: response, isLoading } = useQuery({
+    queryKey: [route.path, options?.pathParams, options?.query],
+    queryFn: async () => {
+      return await request(route, options);
+    },
+    enabled: options?.prefetch ?? true,
+    refetchOnWindowFocus: options?.refetchOnWindowFocus,
+  });
```
Comment (rithviknishad) [inline src/Utils/request/useTanStackQueryInstead.ts]: In this new approach, we are no longer having the ability to abort requests right? We could modify the `request` to take in a `signal` and use the signal provided by tanstack's useQuery's `queryFn`.

```suggestion
export default function useTanStackQueryInstead<TData>(
  route: QueryRoute<TData>,
  options?: QueryOptions<TData>,
) {
  const { data: response, isLoading } = useQuery({
    queryKey: [route.path, options?.pathParams, options?.query],
    queryFn: async ({ signal }) => {
      return await request(route, options, signal);
    },
    enabled: options?.prefetch ?? true,
    refetchOnWindowFocus: options?.refetchOnWindowFocus,
  });
```

**Comment by rithviknishad** [comment]
image

Since we have plugins, this approach isn't backwards compatible. We might have to stick with same file and function name `useQuery` instead of `useTanstackQuery` instead.

**Comment by rithviknishad** [comment]
Looks like there's an issue with refetching when component remounts.

Example: when we go to asset list page, and open an asset and go back to asset list page using the back button, it shows no assets found.

**Comment by rithviknishad** [comment]
Opted out of caching between usages of the useQuery that has the same query key. This was causing so many weird state management issues with our existing implementation.

Cache for the same useQuery invocations is still present as per defaults.

**Review by rithviknishad — APPROVED** [review]

**Comment by bodhish** [comment]
@coderabbitai full review

**Review by Jacobjeevan — APPROVED** [review]
Since we merged Users redesign FE yesterday, there's a few more instances of useQuery that ought to be replaced.

Other than that, LGTM.

**Review by shivankacker — APPROVED** [review]

**Comment by rithviknishad** [comment]
> Since we merged Users redesign FE yesterday, there's a few more instances of useQuery that ought to be replaced.

Done, thanks for finding that miss!

**Pull request merged**